### DISCUSSION TITLE
src/ folder is alternating between Yes/No

### DISCUSSION THREAD

**Discussion by rwieruch** [vercel/next.js#74046] [App Router] [answered]
I recently released a course on Next.js, and some students have reported seeing "No" as the default option when prompted to use the src/ folder during the initialization of a new Next.js project. This has raised questions about why "Yes" isn’t consistently set as the default option.

For example, one student sees "No" as the default option. He is on Node 20.17.0 and npm 10.2.0.

**Comment by icyJoseph** [comment, marked as answer]
> As an alternative to having the special Next.js app or pages directories in the root of your project, Next.js also supports the common pattern of placing application code under the src directory.

- https://nextjs.org/docs/app/building-your-application/configuring/src-directory

Guess it is because `src` directory is considered an alternative? And to add to that, I think the creact-next-app CLI saves your preferences, doesn't it? If you use `src` directory once, the next time around it should be selected as you preference. IIRC.

**Reply by rwieruch** [reply]
This is it. Confirmed by one of my students too :) Thanks for replying here!
